uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/GroupRESTController.java

```diff
@@ -45,8 +47,12 @@ public void setPersonManager(IPersonManager personManager) {
 
     @RequestMapping(value = "/groups", method = RequestMethod.GET)
     public ModelAndView getUsersGroup(HttpServletRequest request, HttpServletResponse response) {
-        Set<Entity> groups =
-                groupService.getGroupsForMember(personManager.getPerson(request).getUserName());
+        IPerson person = personManager.getPerson(request);
+        Set<Entity> groups = Collections.EMPTY_SET;
+        if (person != null) {
+            groups = groupService.getGroupsForMember(person.getUserName());
+        }
+
```

Comment:
GuestUser is created using createGuestPerson method and it has default username assigned to it as "guest". If a person is created/exist, then it already has username.  If you look at the IPerson interface, it has USERNMAE field which has "username" value is assigned to it. That means, person cannot exist without username, so it is unlikely that person.getUserName() will throw NPE. But I will still add that check.